src/main/java/org/folio/circulation/domain/FeeFineRepository.java

```diff
@@ -0,0 +1,50 @@
+package org.folio.circulation.domain;
+
+import static org.folio.circulation.support.http.ResponseMapping.forwardOnFailure;
+import static org.folio.circulation.support.http.ResponseMapping.mapUsingJson;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.client.CqlQuery;
+import org.folio.circulation.support.http.client.PageLimit;
+import org.folio.circulation.support.http.client.Response;
+import org.folio.circulation.support.http.client.ResponseInterpreter;
+
+import io.vertx.core.json.JsonObject;
+
+public class FeeFineRepository {
+  private static final Integer PAGE_LIMIT = 500;
+
+  private final CollectionResourceClient feeFineStorageClient;
+
+  public FeeFineRepository(Clients clients) {
+    feeFineStorageClient = clients.feeFineStorageClient();
+  }
+
+  public CompletableFuture<Result<FeeFine>> getFeeFine(String feeFineOwnerId, String feeFineType) {
+    Result<CqlQuery> typeQuery = CqlQuery.exactMatch("feeFineType", feeFineType);
+    Result<CqlQuery> ownerQuery = CqlQuery.exactMatch("ownerId", feeFineOwnerId);
+
+    return typeQuery.combine(ownerQuery, CqlQuery::and)
+      .after(q -> feeFineStorageClient.getMany(q, PageLimit.limit(PAGE_LIMIT)))
+      .thenApply(r -> r.next(this::mapResponseToFeefines))
+      .thenApply(r -> r.map(MultipleRecords::getRecords))
+      .thenApply(r -> r.map(col -> col.stream().findAny().orElse(null)));
+  }
+
+  private Result<MultipleRecords<FeeFine>> mapResponseToFeefines(Response response) {
+    return MultipleRecords.from(response, FeeFine::from, "feefines");
+  }
+
+  public CompletableFuture<Result<FeeFine>> create(JsonObject feeFineRepresentation) {
```

Comment:
Why does this take JSON instead of the `FeeFine` domain class?

Reply:
Replaced with FeeFine object.